src/v/archival/ntp_archiver_service.cc

```diff
@@ -508,6 +508,48 @@ ss::future<> ntp_archiver::sync_manifest_until_term_change() {
     }
 }
 
+ss::future<> ntp_archiver::unsafe_reset_metadata(ss::input_stream<char> is) {
```

Comment:
Thinking about the overall process, it's going to be dicey to upload a modified manifest while the archiver is still running and could overwrite the new manifest at any time.  In the tickets for this stuff, I had been thinking of the pause operation and the reset operation as working together: first you pause, then you upload a manifest, then you reset + resume.

The real "pause" would be a persistent thing, but maybe for the purposes of debug we can make do with a simple API that uses the _paused mechanism on a running ntp archiver (safe as long as the leadership doesn't change).  Or maybe we just say the caveat to this feature for the moment is that you need to have  the partition in a quiescent state (hard for the user to reason about though, because we can do housekeeping/gc at any time)